code/go/0chain.net/smartcontract/stakepool/stakepool.go

```diff
@@ -189,10 +222,6 @@ func (sp *StakePool) DistributeRewards(
 		return nil
 	}
 
-	if len(sp.Pools) == 0 {
-		return fmt.Errorf("no stake pools to move tokens to")
-	}
-
 	valueLeft := value - serviceCharge
 	var stake = float64(sp.stake())
```

Comment:
I can't get why this sum for all pools equals to 1, `S(pool.Balance / stake)` it can be when all pools balances are equal to stake, is it always correct?

Reply:
```go
var stake = float64(sp.stake())

func (sp *StakePool) stake() (stake state.Balance) {
	for _, pool := range sp.Pools {
		stake += pool.Balance
	}
       return
}
```
And 
```go
	for id, pool := range sp.Pools {
		ratio := float64(pool.Balance) / stake
                reward := state.Balance(valueLeft * ratio)
                ...
	}
```
So combining the two for loops and summing the ratios should give `stake()/stake() = 1`. Does that look ok?

The main difference is the typecasting. In particular the state.Balance cast in the second loop. The alternative here is to keep everything `float64` and store unclaimed rewards less than one ZCN in the StakePool object, rather than randomly being out by +1, 0 or -1 each time.